docs/static-evaluation.md

```diff
@@ -0,0 +1,388 @@
+
+* Introducing an alternate provider address method and updating documentation
+* Provider mappings for modules and resources are hard-coded at config load time on an *unexpanded* view of the config/graph structures
+* Provider configurations are [pruned during graph processing](./provider-references.md#Provider-Workflow)
+
+Potential implementation paths are explored in [Static Evaluation of Providers](./static-evaluation-providers.md)
+
+#### Questions
+
+Should variables be allowed in required_providers now or in the future?  Could help with versioning / swapping out for testing?
+
+In #300, there's also a discussion on allowing variable names in provider aliases.
+Example:
+```
+# Why would you want to do this?  It looks like terraform deprecated this some time after 0.11.
+provider "aws" {
+  alias = var.foo
+}
+```
+
+### Backend Configuration
+
+Once the core is implemented, this is probably the easiest solution to implement.
+
+Notes from initial prototyping:
+* The configs.Backend saves the config body during the config load and does not evaluate it
+* backendInitFromConfig() in command/meta_backend.go is what evaluates the body
+  - This happens before the graph is constructed / evaluated and can be considered an extension of the config loading stage.
+* We can stash a copy of the StaticContext in the configs.Backend and use it in backendInitFromConfig() to provide do the decoding.
+  - There are a few ways to do this, stashing it there was a simple way to get it working in the prototype.
+* Don't forget to update the configs.Backend.Hash() function as that's used to detect any changes
+
+### Lifecycle Attributes
+
+Not yet investigated in depth.
+
+### Variable defaults/validation?
+
+Not sure we are doing this one at this juncture. It may have been passed on before due to complexity around providers, or simply that coalesce + locals exists.
+
+### Provisioners
+
+Not yet investigated in depth.
+
+### Moved blocks
+
+Not yet investigated in depth.
+
+## Encryption
+
+Not yet investigated in depth.
+
+## Blockers:
+### Testing
+
+Existing testing within OpenTofu is fragmented and more sparse than we would like. Additional test coverage will be needed before, during and after many of the following items.
+
+Code coverage should be inspected before refactoring of a component is undertaken to guide the additional test coverage required. We are not aiming for 100%, but should use it as a tool to understand our current testing.
+
+A comprehensive guide on e2e testing should be written, see #1536.
+
+## Unknowns:
+### Providers variables
+Providers may have configuration that depends on variables and dynamic values, such as resources from other providers. There is a odd workaround within the internal/tofu package where variable values may be requested during the graph building phase. This is an odd hack and may need to be reworked for the providers iteration above.
```

Comment:
Can you add a link to where it exists in the code? Also, a link to the documentation discussing it if it exists.